teachopencadd/talktorials/019_md_simulation/talktorial.ipynb

```diff
@@ -0,0 +1,1250 @@
+{

```

Comment:
<p>Note that <code>!command</code>  does not stop the cell from being executed. "Dependencies successfully installed" will be showed regardless the success of the <code>conda install</code>  command. This, in addition to the cleared outputs, can be very misleading... We should check whether things were actually installed or not, maybe try importing openmm or something similar. <em>Only</em> if that succeeds we clear outputs and report success. Otherwise we want the logs to see what went wrong (e.g. the random network errors we saw).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/volkamerlab/TeachOpenCADD/pull/56/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

Reply:
<p>Is it pythonic/ok to nest try except blocks? Something along the lines of</p><pre class="ql-syntax" spellcheck="false">try:
	import condacolab 
    [...]
	try: 
			import rdkit
            clear_output()
            print("Installation successful")
  	except:
      		print("Dependencies not installed")
except: 
  	print("Not on Colab")
</pre><p> </p><p>If not, what would be the better way to check for successful installation? Any <code>If-else-checks</code> possible for installed dependencies?</p>

Reply:
I'd use the `else` clause of the `try` block, which will only be reached if no exception was raised during `try`

```python
try:
    import condacolab 
    ...
except: 
    print("Not on Colab")
else:
    try: 
        import rdkit
        clear_output()
        print("Installation successful")
    except:
        print("Dependencies not installed!")
```

Reply:
<p>Thanks, that is way better</p>